src/features/TakeNow/ui/ValidateTakeNowParams.tsx

```diff
@@ -44,10 +49,21 @@ function ValidateTakeNowParams({
 
   useEffect(() => {
     if (error) {
-      addErrorBanner({
-        children: error,
-        duration: 15000,
-      });
+      let duration: number | null = 15000;
+
+      if (error.type === 'mismatchedRespondent') {
+        logout();
+        duration = null;
+      } else if (error.type === 'invalidApplet') {
+        navigate(ROUTES.appletList.path);
+      }
+
+      setTimeout(() =>
+        addErrorBanner({
+          children: error.error,
+          duration,
+        }),
```

Comment:
@farmerpaul I decided to split the banner fix into a separate PR because I haven't figured out an elegant solution for it yet